Automattic/woocommerce-payments
# Fix incorrect payment method title

Fixes #7272
Fixes #7273 

#### Changes proposed in this Pull Request

This prevents the payment method title from being updated for non-WooPayments gateways and limits those updates to the Legacy UPE gateway.

#### Testing instructions

- Enable Cash On Delivery and WooPayments
- As a customer, add any product to the cart
- Select Cash On Delivery as payment method
- Checkout
- On the Order Confirmation page, ensure the correct payment method title is set

Before:
<img src="https://github.com/Automattic/woocommerce-payments/assets/3473953/ba1836f2-6d2e-4358-a22c-710c8ff46854" width="400" />

After:
<img src="https://github.com/Automattic/woocommerce-payments/assets/3473953/22a6ff14-1734-445a-961d-fad8057823b3" width="400" />














-------------------


**Post merge**
